packages/frontend-2/pages/server-management/active-users.vue

```diff
@@ -0,0 +1,332 @@
+      class="-mt-24 -mb-24"
+      @infinite="infiniteLoad"
+    />
+
+    <UserDeleteDialog
+      v-model:open="showUserDeleteDialog"
+      :user="userToModify"
+      title="Delete User"
+      :buttons="[
+        {
+          text: 'Delete',
+          props: { color: 'danger', fullWidth: true },
+          onClick: deleteConfirmed
+        },
+        {
+          text: 'Cancel',
+          props: { color: 'secondary', fullWidth: true, outline: true },
+          onClick: closeUserDeleteDialog
+        }
+      ]"
+    />
+
+    <ChangeUserRoleDialog
+      v-model:open="showChangeUserRoleDialog"
+      :user="userToModify"
+      title="Change Role"
+      :old-role="oldRole"
+      :new-role="newRole"
+      :hide-closer="true"
+      :buttons="[
+        {
+          text: 'Change Role',
+          props: { color: 'danger', fullWidth: true },
+          onClick: changeUserRoleConfirmed
+        },
+        {
+          text: 'Cancel',
+          props: { color: 'secondary', fullWidth: true, outline: true },
+          onClick: closeChangeUserRoleDialog
+        }
+      ]"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue'
+import { useQuery, useMutation } from '@vue/apollo-composable'
+import { debounce } from 'lodash-es'
+import Table from '~~/components/server-management/Table.vue'
+import UserRoleSelect from '~~/components/server-management/UserRoleSelect.vue'
+import UserDeleteDialog from '~~/components/server-management/DeleteUserDialog.vue'
+import ChangeUserRoleDialog from '~~/components/server-management/ChangeUserRoleDialog.vue'
+import Avatar from '~~/components/user/Avatar.vue'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import { InfiniteLoaderState } from '~~/lib/global/helpers/components'
+import { graphql } from '~~/lib/common/generated/gql'
+import {
+  MagnifyingGlassIcon,
+  ShieldExclamationIcon,
+  ShieldCheckIcon,
+  TrashIcon
+} from '@heroicons/vue/20/solid'
+import { UserItem } from '~~/lib/server-management/helpers/types'
+import {
+  convertThrowIntoFetchResult,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+
+definePageMeta({
+  middleware: ['admin']
+})
+
+const getUsers = graphql(`
+  query UserList($limit: Int!, $cursor: String, $query: String) {
+    admin {
+      userList(limit: $limit, cursor: $cursor, query: $query) {
+        totalCount
+        cursor
+        items {
+          id
+          email
+          avatar
+          name
+          role
+          verified
+          company
+        }
+      }
+    }
+  }
+`)
+
+const adminDeleteUser = graphql(`
+  mutation Mutation($userConfirmation: UserDeleteInput!) {
+    adminDeleteUser(userConfirmation: $userConfirmation)
+  }
+`)
+
+const changeRoleMutation = graphql(`
+  mutation AdminChangeUseRole($userRoleInput: UserRoleInput!) {
+    userRoleChange(userRoleInput: $userRoleInput)
+  }
+`)
+
+const userToModify = ref<UserItem | null>(null)
```

Comment:
`X | null` can be shortened to `Nullable<X>`. `Nullable` comes from `@speckle/shared`. There's also `Optional<X>` for `X | undefined` amongst some other useful utility types in that package.